### DISCUSSION TITLE
INCLUDE query on self referencing nested model.

### DISCUSSION THREAD

**Discussion by smamun19** [prisma/prisma#18665] [Q&A] [answered]
Suppose I have a model like this  

```
model Category {
  id          Int         @id @default(autoincrement())
  name        String
  isSub       Boolean     @default(false)
  subCategory Category[]  @relation("sub_category")
  category    Category?   @relation("sub_category", fields: [categoryId], references: [id])
  categoryId  Int?
}
 ```

How can I fetch all the nested child category from this relation?

I tried the following query

```
  prisma.category.findMany({
      where: { isSub: false },
      include: { subCategory: true },
    });
```
But this query results in only  1 level of nested data.

**Comment by nurul3101** [comment, marked as answer]
Hello @smamun19 👋 

Prisma currently does not support fetching all deeply nested relations in a single query.

We have a feature request for it here:
- https://github.com/prisma/prisma/issues/3725

For now, you would need to use a recursive function, as mentioned in the above feature request.

**Reply by nurul3101** [reply]
If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by smamun19** [reply]
Hello @nurul3101!
Sorry for the late reply. Unfortunately I haven't had the time to check on the issue yet. I will close the Discussion after checking on the answer within few days :slightly_smiling_face:

**Reply by nurul3101** [reply]
No worries!
Thanks for getting back and letting us know! :)

**Comment by solruh** [comment]
if we know the depth(level of children we want), wouldn't it be okay to do something like below?
```
await prisma.category.findMany({
  include: {
    subCategory: {
      include: {
        subCategory: {
          include: {
            subCategory: true //we want all children up to this depth
          }
        }
      }
    }
  }
});
```
